Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1085,6 +1082,58 @@ private extension PurchasesOrchestrator {
         }
     }
 
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    private func syncPurchasesSK2(isRestore: Bool,
+                                  initiationSource: ProductRequestData.InitiationSource,
+                                  completion: (@Sendable (Result<CustomerInfo, PurchasesError>) -> Void)?) {
+        let currentAppUserID = self.appUserID
+        let unsyncedAttributes = self.unsyncedAttributes
+
+        self.attribution.unsyncedAdServicesToken { adServicesToken in
+            _ = Task<Void, Never> {
+                let transaction = await self.transactionFetcher.firstVerifiedAutoRenewableTransaction
+                guard let transaction = transaction, let jwsRepresentation = transaction.jwsRepresentation  else {
+                    self.customerInfoManager.customerInfo(appUserID: currentAppUserID,
+                                                          fetchPolicy: .cachedOrFetched) { result in
+                        switch result {
+                        case .success(let customerInfo):
+                            self.operationDispatcher.dispatchOnMainThread {
+                                completion?(.success(customerInfo))
+                            }
+                        case .failure(let error):
+                            self.operationDispatcher.dispatchOnMainThread {
+                                completion?(.failure(ErrorUtils.customerInfoError(error: error)))
+                            }
+                        }
```

Comment:
This is redundant, it can be simplified in a single line:
```suggestion
                                completion(result.mapError(\.asPurchasesError))
```